data_structures/src/chain.rs

```diff
@@ -2246,14 +2248,19 @@ fn update_utxo_outputs(
     outputs: &[ValueTransferOutput],
     txn_hash: Hash,
 ) {
-    for (index, output) in outputs.iter().enumerate() {
-        // Add the new outputs to the utxo_set
-        let output_pointer = OutputPointer {
-            transaction_id: txn_hash,
-            output_index: u32::try_from(index).unwrap(),
-        };
+    let mut index = 0;
+    for output in outputs.iter() {
+        for i in 0..output.multiplier {
+            // Add the new outputs to the utxo_diff
+            let output_pointer = OutputPointer {
+                transaction_id: txn_hash,
+                output_index: u32::try_from(index + i).unwrap(),
```

Comment:
I may be missing something here, but why is this `try_from` needed under the assumption that `output.multiplier` is `u32` and therefore `i` and `index` have their type derived from it?